File: wrappers/java/src/main/java/org/w21parser/W21Exception.java
package org.w21parser;

public class W21Exception extends Exception {
    public final int error;
    private final String faultstring;
    private final String XMLfaultdetail;
    W21Exception(
            String message, int error,
            String faultstring, String XMLfaultdetail
    ) {
        super(message);
        this.error = error;
        this.faultstring = (faultstring != null)?faultstring:"";
        this.XMLfaultdetail = (XMLfaultdetail != null)?XMLfaultdetail:"";
    }

    public String getFaultstring() { return this.faultstring; }
    public String getXMLfaultdetail() { return this.XMLfaultdetail; }
}

File: wrappers/dotNET/W21CSparser/exceptions/W21Exception.cs
namespace W21CSparser.exceptions;

public class W21Exception : Exception
{
    public int Error { get; }
    public string FaultString { get; }
    public string XmlFaultDetail { get; }

    public W21Exception(string message, int error, string? faultstring, string? xmlfaultdetail) 
        : base(message)
    {
        Error = error;
        
        FaultString = faultstring ?? string.Empty;
        XmlFaultDetail = xmlfaultdetail ?? string.Empty;
    }
}